### ISSUE TITLE
Pagination starts at 0

### ISSUE THREAD

**Issue by Fronix** [meilisearch/instant-meilisearch#1075]
**Description**
Pagination is starting at 0, I don't really understand the reason. I've never seen pagination have 0,1,2,3.
It's the same for both finite and non finite pagination.

**Expected behavior**
I expect instant-meilisearch to return the same page information as meilisearch, if I do an initial search and meilisearch returns
```
page: 1
totalHits: 1000
totalPages: 20
```

I expect the pagination to reflect this

```
nbPages: 20
currentRefinement: 1
```

**Current behavior**
If I load the search page, my pagination component returns, 
```
{
    "canRefine": true,
    "currentRefinement": 0,
    "nbHits": 1000,
    "nbPages": 20,
    "pages": [
        0,
        1,
        2,
        3,
        4,
        5,
        6
    ],
    "isFirstPage": true,
    "isLastPage": false
}
```

**Environment (please complete the following information):**
- Meilisearch version: v1.1
- instant-meilisearch version: v0.12.0

**Comment by Fronix** [comment]
> Hey @Fronix, is it showing `0` on the UI components ? InstantSearch starts their pagination at `0`, so I need to return `0` when we are on the first page or `- 1` in other further pages. If instantMeilisearch returns `1` for the first page instead of `0`, the UI components would start at page 2.

I see, this might be a InstanSearch problem then. The issue starts when we have components like MUI `Pagination` which assumes pagination starts with 1. So when you load the page, `currentRefinment` is 0 but `nbPages` is, lets say 20. The component assumes the `nbPages` value starts at 1, so you have to reinvent the wheel again to harmonize 0 to 1 and all other pages after that.

It just felt so odd that meilisearch starts at 1, the MUI pagination component assumes it starts with 1 and then instantsearch starts with 0?

**Comment by bidoubiwa** [comment]
I agree, unfortunately I cannot provide a solution to that :( Maybe you can try creating an issue on instantsearch's repo?
